jota/src/main/java/jota/utils/Signing.java

```diff
@@ -131,7 +131,7 @@ public Signing(ICurl curl) {
     }
 
     public int[] digest(int[] normalizedBundleFragment, int[] signatureFragment) {
-        curl.reset();
+    	ICurl iCurl = this.evaluateDigestCurlObject();
         ICurl jCurl = this.getICurlObject(SpongeFactory.Mode.KERL);
```

Comment:
This should also be not KERL but a clone of iCurl.

Reply:
@schierlm Explain me the reason behind, because if you see the history of this class "Signing.class" you can see that it always has used an instance of type Kerl to make an operation in the inner for structure of the digest method.

Reply:
@schierlm If you see the iCurl object of type Kerl it can implement distinct operations depending on the specific kind of object. It looks like logic to clone the iCurl object instead of the jCurl object but some implementations details can exist behind.

Reply:
The idea of passing your own digest (Curl or Kerl) is to be able to generate both old addresses before the transition as the [oldSigning](https://github.com/iotaledger/iota.lib.js/blob/master/lib/crypto/signing/oldSigning.js#L122) class of `iota.lib.js` does, as well as new addresses after the transition (as the [signing](https://github.com/iotaledger/iota.lib.js/blob/master/lib/crypto/signing/signing.js#L125) class of `iota.lib.js` does. When you look closely at **oldSigning** you will see that it uses 2 Curl instances where new **signing** class uses 2 Kerl instances. Therefore, current Java version cannot generate the same addresses as **oldSigning** class of `iota.lib.js`.

If this is intentional, I'd suggest to remove the functionalty to pass in a Curl instance, as it confuses the users when they don't get the same addresses.

Reply:
And it is not true that it never changed. It changed [here](https://github.com/iotaledger/iota.lib.java/commit/f3a32d4bdc4eb8a9cf4523a04918ed15940fbde7#diff-9d0c782720089dc2e669daf55a02428aR149).